lms/resources/lti_launch.py

```diff
@@ -197,6 +198,30 @@ def canvas_groups_enabled(self):
 
         return bool(application_instance.settings.get("canvas", "groups_enabled"))
 
+    @property
+    def blackboard_groups_enabled(self):
+        """Return True if BlackBoard groups are enabled at the school/installation level."""
+        try:
+            application_instance = self._application_instance_service.get_current()
+        except ApplicationInstanceNotFound:
+            return False
+
+        return bool(application_instance.settings.get("blackboard", "groups_enabled"))
+
+    @property
+    def is_blackboard_group_launch(self):
+        """Return True if the current assignment uses BlackBoard groups."""
+        if not self.blackboard_groups_enabled:
+            return False
+
+        tool_consumer_instance_guid = self._request.params[
+            "tool_consumer_instance_guid"
+        ]
+        assignment = self._assignment_service.get(
```

Comment:
I had this as an `assignment` property here and also proposed having a `get_current` on the service itself but at the end inlined it as it depends on `self.resource_link_id` which it made it adding it on the service too coupled to not only the request but an specific context.